### PR TITLE
Added support for quota expression

### PR DESCRIPTION
**Pull request by gustavo-iniguez-goya**
Hi!

This PR adds support for quotas.

### CODE REVIEW

**Comment by stapelberg** [comment]
Thanks for your PR! Could you add a corresponding test function to `nftables_test.go` as well please?

**Comment by gustavo-iniguez-goya** [comment]
Added! sorry, I had some problems figuring out how to extract the want-ed bytes for the tests.
While debugging it I've realized that the flags attribute must be added unconditionally so I've updated quota.go accordingly.

File: expr/quota.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	"github.com/google/nftables/binaryutil"
+	"github.com/mdlayher/netlink"
+	"golang.org/x/sys/unix"
+)
+
+// Quota defines a threshold against a number of bytes.
+type Quota struct {
+	Bytes uint64
+	Used  uint64
```
Comment (stapelberg) [inline expr/quota.go:28]: Should this field be named `Consumed` instead? That would match terminology on the kernel side at least.

File: nftables_test.go [hunk 2]
```diff
@@ -4768,6 +4768,77 @@ func TestNotrack(t *testing.T) {
...
+		Table: filter,
+	})
+
+	c.AddRule(&nftables.Rule{
+		Table: filter,
+		Chain: output,
+		Exprs: []expr.Any{
+			// [ quota bytes 6 consumed 0 flags 1 ]
+			&expr.Quota{
+				Bytes: uint64(6),
```
Comment (stapelberg) [inline nftables_test.go:4830]: Here and in the next line, I don’t think you need the explicit `uint64(6)`, and just using `6` should work, no?

**Review by stapelberg — CHANGES_REQUESTED** [review]

File: expr/quota.go (hunk 1 above)
Comment (gustavo-iniguez-goya) [reply expr/quota.go]: yes, you're right. I took "used" from the output of `list ruleset`.

**Review by stapelberg — APPROVED** [review]

**Comment by stapelberg** [comment]
Thanks!

**Pull request merged**